extensions/amp-samba-player/0.1/amp-samba-player.js

```diff
@@ -0,0 +1,260 @@
+import {getDataParamsFromAttributes,removeElement} from '../../../src/dom';
+import {
+  installVideoManagerForDoc,
+} from '../../../src/service/video-manager-impl';
+import {videoManagerForDoc} from '../../../src/video-manager';
+import {VideoEvents} from '../../../src/video-interface';
+import {tryParseJson} from '../../../src/json';
+import {user} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-samba-player';
+
+/** @private {Object} */
+const API_MAP = {
+  'web4-7091': '//playerapitest2.liquidplatform.com:7091/embed/',
+  'localhost-8080': '//localhost:8080/player-api/embed/',
+  'web4-7021': '//playerapitest2.liquidplatform.com:7021/embed/',
+  'staging': '//staging-player-api.sambavideos.sambatech.com/v3/embed/',
+  'prod': '//fast.player.liquidplatform.com/pApiv2/embed/',
+};
+
+/** @private {Object} */
+const EVENTS_MAP = {
+  'onPlay': VideoEvents.PLAY,
+  'onPause': VideoEvents.PAUSE,
+  'onMute': VideoEvents.MUTED,
+  'onUnmute': VideoEvents.UNMUTED,
+};
+
+class AmpSambaPlayer extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {string} */
+    this.projectId_ = '';
+
+    /** @private {string} */
+    this.mediaId_ = '';
+
+    /** @private {string} */
+    this.env_ = 'prod';
+
+    /** @private {string} */
+    this.protocol_ = this.win.location.protocol || 'http';
+
+    /** @private {Object} */
+    this.params_ = {};
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+
+    /** @private {?Function} */
+    this.playerLoadResolver_ = null;
+
+    /** @private {Promise} */
+    this.playerLoadPromise_ = new Promise(resolve => {
+      this.playerLoadResolver_ = resolve;
+    });
+
+    // Event handlers bindings
+    this.playerMessageHandler_ = this.playerMessageHandler_.bind(this);
+  }
+
+  /** @override */
+  preconnectCallback(opt_onLayout) {
+    // host to serve the player
+    this.preconnect.url(API_MAP['prod'], opt_onLayout);
+    // host to serve media contents
+    this.preconnect.url('http://pvbps-sambavideos.akamaized.net', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    // support all layouts
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.projectId_ = user().assert(
+      this.element.getAttribute('data-project-id'),
+      `The data-project-id attribute is required for <${TAG}> %s`,
+      this.element);
+
+    // not required in case of live
+    this.mediaId_ = this.element.getAttribute('data-media-id') || '';
+
+    // which environment to run
+    this.env_ = this.element.getAttribute('data-env') || this.env_;
+
+    // the protocol to be used (HTTP, HTTPS)
+    this.protocol_ = (this.element.getAttribute('data-protocol')
+      || this.protocol_).replace(/\:?$/, ':');
+
+    // player features related params
+    // WARN: methods missing on returned object (e.g. hasOwnProperty) so recreate it
```

Comment:
That's by design, why do you need access to `hasOwnProperty`? `getDataParamsFromAttributes` make a pure map so you can do `for in` without needing to check `hasOwnProperty`